extractor/src/test/java/com/example/extractor/MainTest.java

```diff
@@ -27,157 +27,106 @@ public void testParseConfigFromHeader() throws Exception {
-        String outputFilePath = "src/test/resources/generated-config-example1.xml";
         String expectedFilePath = "src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/configs/config-example1-expected.xml";
 
-        ConfigSerializer.serializeConfig(exampleFilePath, templateFilePath, outputFilePath);
-
-        String generatedContent = new String(Files.readAllBytes(Path.of(outputFilePath)));
-        String expectedContent = new String(Files.readAllBytes(Path.of(expectedFilePath)));
+        String generatedContent = ConfigSerializer.serializeConfigToString(exampleFilePath, templateFilePath);
+        String expectedContent = ConfigSerializer.sortProperties(
+                new String(Files.readAllBytes(Path.of(expectedFilePath)))
+        );
 
-        // Sort the properties in both the generated and expected content
-        String sortedGeneratedContent = ConfigSerializer.sortProperties(generatedContent);
-        String sortedExpectedContent = ConfigSerializer.sortProperties(expectedContent);
-
-        assertEquals(sortedExpectedContent.trim(), sortedGeneratedContent.trim(), "The generated configuration does not match the expected output.");
+        assertThat(generatedContent.trim()).isEqualTo(expectedContent.trim());
     }
 
     @Test
     public void testGeneratedConfigMatchesExpectedExample2() throws Exception {
         String exampleFilePath = "src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/examples/Example2.java";
         String templateFilePath = "src/main/resources/config-template-treewalker.xml";
-        String outputFilePath = "src/test/resources/generated-config-example2.xml";
         String expectedFilePath = "src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/configs/config-example2-expected.xml";
 
-        ConfigSerializer.serializeConfig(exampleFilePath, templateFilePath, outputFilePath);
-
-        String generatedContent = new String(Files.readAllBytes(Path.of(outputFilePath)));
-        String expectedContent = new String(Files.readAllBytes(Path.of(expectedFilePath)));
-
-        // Sort the properties in both the generated and expected content
-        String sortedGeneratedContent = ConfigSerializer.sortProperties(generatedContent);
-        String sortedExpectedContent = ConfigSerializer.sortProperties(expectedContent);
+        String generatedContent = ConfigSerializer.serializeConfigToString(exampleFilePath, templateFilePath);
+        String expectedContent = ConfigSerializer.sortProperties(
+                new String(Files.readAllBytes(Path.of(expectedFilePath)))
+        );
 
-        assertEquals(sortedExpectedContent.trim(), sortedGeneratedContent.trim(), "The generated configuration does not match the expected output.");
+        assertThat(generatedContent.trim()).isEqualTo(expectedContent.trim());
```

Comment:
```
String generatedContent = ConfigSerializer.serializeConfigToString(exampleFilePath, templateFilePath);
String expectedContent = loadToString(expectedFilePath);
assertThat(generatedContent).isEqualTo(expectedContent);
```

Reply:
What would loadToString() do?
So, right now, generated content goes through follownig process: 1. Parse from Examples# 2. Extact relvant moduleName and properties header 3. Sort it. 4. Replace them in the config-template. 


Given my understanding, you also want this to be done for expected files? "We should extract from ExampleX, sort properties, save to file and name it "-expected", store it to resources." - as mentioned above. Is that what loadToString expected to do?

Reply:
>loadToString()

Just load any file to string

Reply:
Expected file as loaded only as text files, no parsing.

Reply:
I am still not clear on what loadToString does. Can you give me some more context?

Reply:
Also, we are not sorting properties. How would it pass the aseertTrue?

Reply:
`new String(Files.readAllBytes(Path.of(expectedFilePath))) `Wasn't it loading a file to String?